store/src/java/com/zimbra/cs/mailbox/Mailbox.java

```diff
@@ -10494,14 +9712,8 @@ private boolean referencesOtherMailbox(ItemIdentifier ident) {
 
     /** Acquire an in process lock relevant for this type of MailboxStore */
     @Override
-    public void lock(boolean write) {
-        lock.lock(write);
-    }
-
-    /** Release an in process lock relevant for this type of MailboxStore */
-    @Override
-    public void unlock() {
-        lock.release();
+    public MailboxLock lock(boolean write) {
```

Comment:
This doesn't lock while the old one did.

Reply:
I changed the semantics of this from "do lock" to "return lock". I probably could have complicated the implementation of `LockFactory` so that each `LockFactory` would be responsible for keeping track of any produced locks so that the caller of `lock` would not have to track the lifecycle of the lock. But the code is more readable if `lock` returns a `AutoClosable` lock object that can be used with `try-with-resources`